app/chronology/models.py

```diff
@@ -17,41 +20,114 @@ def default_festival_settings() -> Dict[str, EmptyDict]:
     return {"theme": {}, "forms": {}}
 
 
-class Festival(models.Model):
-    end_time = models.DateTimeField(verbose_name=_("end time"))
+class Festival(ComputedFieldsModel):
+
+    DRAFT = "draft"
+    READY = "ready"
+    PROPOSAL = "proposal"
+    AGENDA = "agenda"
+    ONGOING = "ongoing"
+    PAST = "past"
+    STATUS_CHOICES = (
+        (DRAFT, _("draft")),
+        (READY, _("ready")),
+        (PROPOSAL, _("proposal")),
+        (AGENDA, _("agenda")),
+        (ONGOING, _("ongoing")),
+        (PAST, _("past")),
+    )
+
+    start_time = models.DateTimeField(
+        blank=True, null=True, verbose_name=_("start time")
+    )
+    end_time = models.DateTimeField(blank=True, null=True, verbose_name=_("end time"))
     history = HistoricalRecords(
         table_name="ch_festival_history",
         history_change_reason_field=models.TextField(null=True),
+        bases=[ComputedFieldsModel],
     )
     name = models.CharField(max_length=255, verbose_name=_("name"))
     settings = JSONField(default=default_festival_settings, verbose_name=_("settings"))
     slug = models.SlugField(verbose_name=_("slug"))
     sphere = models.ForeignKey(
         Sphere, on_delete=models.CASCADE, verbose_name=_("sphere")
     )
-    start_proposal = models.DateTimeField(verbose_name=_("start proposal"))
-    start_publication = models.DateTimeField(verbose_name=_("start publication"))
-    start_time = models.DateTimeField(verbose_name=_("start time"))
+    start_proposal = models.DateTimeField(
+        blank=True, null=True, verbose_name=_("start proposal")
+    )
+    start_publication = models.DateTimeField(
+        blank=True, null=True, verbose_name=_("start publication")
+    )
 
     class Meta:
         db_table = "ch_festival"
         verbose_name = _("festival")
         verbose_name_plural = _("festivals")
+        constraints = [
+            models.UniqueConstraint(
+                fields=("sphere", "slug"), name="festival_has_unique_slug_and_sphere"
+            ),
+            models.CheckConstraint(
+                check=Q(
+                    start_proposal__isnull=True,
+                    start_publication__isnull=True,
+                    start_time__isnull=True,
+                    end_time__isnull=True,
+                )
+                | Q(
+                    start_proposal__lte=F("start_publication"),
+                    start_publication__lte=F("start_time"),
+                    start_time__lt=F("end_time"),
+                ),
+                name="festival_date_times",
+            ),
+        ]
 
     def __str__(self) -> str:
         return self.name
 
+    @computed(
```

Comment:
Impressive, but could it be just `@property`?

Reply:
It won't be save in database then